### DISCUSSION TITLE
Batch-Replace older vehicles with when improved versions are available

### DISCUSSION THREAD

**Discussion by dwoydig** [voxeltycoon/issues#542] [UI&UX] [answered]
This is something that is missing in a lot of similar games:
When a new truck is available, there should be function in the fleet management window to replace all vehicles of the old, 5t capacity truck wood truck with the new 7t capacity wood truck, and the 5t ore truck with the 7t ore truck.

Right now this is a lot of work, there is no "replace vehicles" function (for lines or individual cars/trains) at all and you have to send to depot, sell, buy, reassign routes and release from garage.

**Comment by victorleblais** [comment, marked as answer]
This feature is planned
![image](https://user-images.githubusercontent.com/4365075/115121388-c56a3880-9fb2-11eb-9033-37b8548ee74d.png)
Check the roadmap here: https://trello.com/b/wHz5JHqH/voxel-tycoon-roadmap

I'm leaving this issue opened so that people can still suggest how this feature could work.

**Comment by asiaclonk** [comment]
Once the "trucks don't teleport to depot" feature is complete, I imagine you can tell vehicles to change into another type of your choice and it will queue a trip to the nearest depot (or a depot of your choice) after its current route loop is done. Once in the depot it will immediate sell and buy into the new vehicle and resume the route.
If your funds would not be enough to buy the upgrade after selling the old vehicle at the time you issued the order, deny it. If your funds decreased after the vehicle started returning and wouldn't be able to afford the upgrade once inside the depot, cancel there and set the vehicle to idle or dispatch it again.
